postprocessing/comstockpostproc/comstock.py

```diff
@@ -147,48 +148,90 @@ def __init__(self, s3_base_dir, comstock_run_name, comstock_run_version, comstoc
-            self.add_energy_intensity_columns()
-            self.add_bill_intensity_columns()
-            self.add_energy_rate_columns()
-            self.add_normalized_qoi_columns()
-            self.add_vintage_column()
-            self.add_dataset_column()
-            # self.add_upgrade_building_id_column()  # TODO POLARS figure out apply function
-            self.add_hvac_metadata()
-            self.add_building_type_group()
-            self.data = self.reduce_df_memory(self.data)
-            self.add_enduse_fuel_group_columns()
-            self.add_enduse_group_columns()
-            self.add_addressable_segments_columns()
-            self.combine_emissions_cols()
-            self.add_metadata_index_col()
-            self.get_comstock_unscaled_monthly_energy_consumption()
+            up_lazyframes = []
+            upgradIdcount = {}
+            upgrade_ids.sort()
+            for upgrade_id in upgrade_ids:
+                self.data = None
+                # Change this to only load 1 upgrade
+                self.load_data(upgrade_id, acceptable_failure_percentage, drop_failed_runs)
+                self.add_buildstock_csv_columns()
+                self.add_geospatial_columns()  # TODO remove geospatial join once reliably in buildstock.csv
+                self.add_ejscreen_columns()
+                self.add_cejst_columns()
+                self.data = self.downselect_imported_columns(self.data)
+                self.rename_columns_and_convert_units()
+                self.set_column_data_types()
+                # Calculate/generate columns based on imported columns
+                # self.add_aeo_nems_building_type_column()  # TODO POLARS figure out apply function
+                self.add_missing_energy_columns()
+                self.combine_utility_cols()
+                self.add_enduse_total_energy_columns()
+                self.add_energy_intensity_columns()
+                self.add_bill_intensity_columns()
+                self.add_energy_rate_columns()
+                self.add_normalized_qoi_columns()
+                self.add_vintage_column()
+                self.add_dataset_column()
+                # self.add_upgrade_building_id_column()  # TODO POLARS figure out apply function
+                self.add_hvac_metadata()
+                self.add_building_type_group()
+                self.data = self.reduce_df_memory(self.data)
+                self.add_enduse_fuel_group_columns()
+                self.add_enduse_group_columns()
+                self.add_addressable_segments_columns()
+                self.combine_emissions_cols()
+
+                if upgrade_id not in upgradIdcount:
+                    upgradIdcount[upgrade_id] = self.data.filter(pl.col(self.UPGRADE_ID) == upgrade_id).shape[0] 
 
+                self.add_metadata_index_col(upgradIdcount)
+                self.get_comstock_unscaled_monthly_energy_consumption()
+                self.add_weighted_energy_savings_columns()
```

Comment:
This method requires the `self.BLDG_WEIGHT` column to exist...but that column is not added until `self.add_national_scaling_weights()` is called. How is the code working now?